src/main/java/eu/decentsoftware/holograms/api/listeners/PlayerListener.java

```diff
@@ -53,6 +64,89 @@ public void onRespawn(PlayerRespawnEvent e) {
         S.async(() -> decentHolograms.getHologramManager().hideAll(player));
     }
 
+
+    // 视角对着的全息
+    private Map<Player,String> viewHolograms = new ConcurrentHashMap<>();
+
+
+    private Map<Player,Long> tickLog = new ConcurrentHashMap<>();
+
+    public Map<Chunk,List<Hologram>> chunkListMap = new ConcurrentHashMap<>();
+
+    private void ref(){
+        chunkListMap.clear();
+        decentHolograms.getHologramManager().getHolograms().stream().forEach(hologram -> {
+            Chunk chunk = hologram.getLocation().getChunk();
+            chunkListMap.putIfAbsent(chunk,new ArrayList<>());
+            chunkListMap.get(chunk).add(hologram);
+        });
+    }
+
+
+    @EventHandler
+    public void onRegisterHd(DecentHologramsRegisterEvent registerEvent){
+        this.ref();
+    }
+
+    @EventHandler
+    public void onPlayerMove(PlayerMoveEvent playerMoveEvent){
+        Player player = playerMoveEvent.getPlayer();
+        Long orDefault = tickLog.getOrDefault(player, 0L);
+
+        if(chunkListMap.isEmpty()) ref();
+
+        if((System.currentTimeMillis()-orDefault) >= 200L){
+            Optional.ofNullable(chunkListMap.get(player.getLocation().getChunk())).ifPresent(list->{
+                List<Hologram> holograms = list.stream().collect(Collectors.toList());
+                Vector playerDirection = player.getEyeLocation().getDirection();
+                double threshold = 0.2; // 0.2弧度
+
+                tickLog.put(player,System.currentTimeMillis());
+
+                holograms.removeIf(hologram -> {
+                    Location location = hologram.getLocation();
+                    Vector toHologram = location.toVector().subtract(player.getEyeLocation().toVector());
+                    double angle = playerDirection.angle(toHologram);
+                    return angle>=threshold;
+                });
+
+                String currentViewHd = viewHolograms.get(player);
+
+
+                for (Hologram hologram : holograms) {
+                    if(currentViewHd==null) viewHolograms.put(player,hologram.getName());
+
+                    currentViewHd = viewHolograms.get(player);
+
+                    // 如果从一个全息移动到了另一个全息上
+                    if(!currentViewHd.equalsIgnoreCase(hologram.getName())){
+                        Hologram fromHologram = decentHolograms.getHologramManager().getHologram(currentViewHd);
+                        List<Integer> clickableEntityIds = fromHologram.getPage(player).getClickableEntityIds();
+                        if(clickableEntityIds.size()>0){
+                            fromHologram.onClick(player,clickableEntityIds.get(0),ClickType.HOVER_LEAVE);
```

Comment:
It would be better to have dedicated onHover methods.
Maybe with a boolean to set whether they look on it or not (i.e. `onHover(id, boolean)`). The click type feels wrong here.